File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/ConnectionErrorEvent.java
package dev.robocode.tankroyale.botapi.events;

import java.net.URI;

/**
 * Event occurring when a connection error occurs.
 */
@SuppressWarnings("unused")
public final class ConnectionErrorEvent extends ConnectionEvent {

    // The error.
    private final Throwable error;

    /**
     * Initializes a new instance of the ConnectionErrorEvent class.
     *
     * @param serverUri is the URI of the server.
     * @param error     is the error.
     */
    public ConnectionErrorEvent(URI serverUri, Throwable error) {
        super(serverUri);
        this.error = error;
    }

    /**
     * Returns the error.
     *
     * @return The error.
     */
    public Throwable getError() {
        return error;
    }
}

File: bot-api/dotnet/Robocode.TankRoyale.BotApi/src/events/ConnectionErrorEvent.cs
using System;
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when a connection error occurs.
/// </summary>
public sealed class ConnectionErrorEvent : ConnectionEvent
{
    /// <summary>The exception causing the error.</summary>
    public Exception Exception { get; }

    /// <summary>
    /// Initializes a new instance of the ConnectionErrorEvent class.
    /// </summary>
    /// <param name="serverUri">URI of the server.</param>
    /// <param name="exception">Exception causing the error.</param>
    [JsonConstructor]
    public ConnectionErrorEvent(Uri serverUri, Exception exception) : base(serverUri) => Exception = exception;
}